lib/setup-lib.bash

```diff
@@ -168,14 +150,50 @@ source "\$(asdf direnv hook asdf)"
 EOF
 }
 
+function print_usage() {
+  echo "Usage: asdf direnv setup [--shell SHELL] [--version VERSION]"
+  echo ""
+  echo "SHELL: one of bash, zsh, or fish. If not specified, defaults to $shell"
+  echo "VERSION: one of system, latest, or x.y.z"
+}
+
 function setup_command() {
-  if test -n "$1"; then
-    export ASDF_DIRENV_SHELL="$1"
+  local shell="$SHELL"
+  local version=""
+
+  while [[ $# -gt 0 ]]; do
+    arg=$1
+    shift
+    case $arg in
+      -h | --help)
+        print_usage
+        exit 1
+        ;;
+      --shell)
+        shell="$1"
+        shift
+        ;;
+      --version)
+        version="$1"
+        shift
+        ;;
+      *)
+        echo "Unknown option: $arg"
+        exit 1
+        ;;
+    esac
+  done
+
+  if [ -z "$version" ]; then
+    echo "Please specify a version using --version"
```

Comment:
does it makes sense to have a default ?

Reply:
Yeah, I really don't know. Like I said above, I'd kind of like to steer our users towards `system` installs of direnv and maybe someday drop support for installing direnv entirely.

So I think my preference would actually be to take a step in that direction and make "system" the default installation type.

Curious what you think, @vic?